### PR TITLE
Fix UnauthorizedAccessException in GetDrives

### PR DESCRIPTION
**Pull request by Bond-009**
```
[15:01:24] [ERR] [55] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL GET /Environment/Drives.
System.UnauthorizedAccessException: Access to the path is denied.
 ---> System.IO.IOException: Operation not permitted
   --- End of inner exception stack trace ---
   at System.IO.DriveInfo.CheckStatfsResultAndThrowIfNecessary(Int32 result)
   at System.IO.DriveInfo.get_TotalSize()
   at Emby.Server.Implementations.IO.ManagedFileSystem.<>c.<GetDrives>b__32_0(DriveInfo d) in /home/bond/dev/jellyfin/Emby.Server.Implementations/IO/ManagedFileSystem.cs:line 583
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.ToList()
   at Emby.Server.Implementations.IO.ManagedFileSystem.GetDrives() in /home/bond/dev/jellyfin/Emby.Server.Implementations/IO/ManagedFileSystem.cs:line 583
   at Jellyfin.Api.Controllers.EnvironmentController.GetDrives() in /home/bond/dev/jellyfin/Jellyfin.Api/Controllers/EnvironmentController.cs:line 153
   at lambda_method559(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
```

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 1]
```diff
@@ -580,7 +580,11 @@ public virtual List<FileSystemMetadata> GetDrives()
         {
             // check for ready state to avoid waiting for drives to timeout
             // some drives on linux have no actual size or are used for other purposes
-            return DriveInfo.GetDrives().Where(d => d.IsReady && d.TotalSize != 0 && d.DriveType != DriveType.Ram)
+            return DriveInfo.GetDrives()
+                .Where(
+                    d => (d.DriveType == DriveType.Fixed || d.DriveType == DriveType.Network || d.DriveType == DriveType.Removable)
```
Comment (cvium) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs:585]: Which types are causing issues?

Comment (Bond-009) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs:585]: `Unknown`

Comment (cvium) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs:585]: Shouldn't the check then be `d.DriveType != DriveType.Unknown`? You're filtering out a bunch of types that should work (admittedly RAMDisk is stupid, but that's neither here nor there).

Comment (Bond-009) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs:585]: We already filtered out ram

**Review by cvium — APPROVED** [review]

**Pull request merged**